src/Gizmos/gizmo.ts

```diff
@@ -331,6 +332,23 @@ export class Gizmo implements IDisposable {
                 lmat.copyFrom(bone.getWorldMatrix());
             }
             bone.markAsDirty();
+        } else if (this._attachedNode.getClassName() === "SpotLight" || this._attachedNode.getClassName() === "PointLight" || this._attachedNode.getClassName() === "DirectionalLight") {
+            var light = this._attachedNode as ShadowLight;
+            const parent = light.parent;
+
+            if (parent) {
+                var invParent = this._tempMatrix1;
+                var nodeLocalMatrix = this._tempMatrix2;
+                parent.getWorldMatrix().invertToRef(invParent);
+                light.getWorldMatrix().multiplyToRef(invParent, nodeLocalMatrix);
+                nodeLocalMatrix.decompose(undefined, this._tempQuaternion, this._tempVector);
+            } else {
+                this._attachedNode._worldMatrix.decompose(undefined, this._tempQuaternion, this._tempVector);
+            }
+            // setter doesn't copy values. Need a new Vector3
+            light.position = new Vector3(this._tempVector.x, this._tempVector.y, this._tempVector.z);

```

Comment:
you can use .copyFrom instead of creating a new vector

Reply:
Nop. That's what was done in LightGizmo and setter is not called...and dirty flag was unchanged. Or I missed something?

Reply:
We don't need the setter if position is set. You use the getter to get the vector and update it. Unless you need the setter to be called?

Reply:
yes, I need the setter to be called.

Reply:
if there is no way around it it's ok, of course :-)
If it's called on each frame i would prefer adding a "setDirty" function to the light so that you can set it yourself (because of gc), but if it's not the case we can leave it this way

Reply:
well, there are pros and cons for setDirty. As it's an internal state, it's awkward to force its state to recompute that way. That's a debate for another day :)